File: 6th Semester/Programming Paradigms/NP2 Exercises - Part 2/ex24.c
// – Modifique o programa anterior de forma a ler um numero n. Imprima os n primeiros números primos.

#include <stdio.h>

int ehPrimo(int num)
{
    int i = 1;
    do
    {
        if ((i != 1 && num % i == 0) || num < 2 || (num != 2 && num % 2 == 0))
        {
            return 0;
        }
        i += 2;
    } while (i <= (num / 2) + 1);
    return 1;
}
int main()
{
    int num;
    printf("Digite até quanto se quer conferir os números primos: ");
    scanf("%i", &num);
    num = num < 0 ? -num : num;
    for (int i = 0; i < num; i++)
        if (ehPrimo(i))
            printf("%d ", i);
    return 0;
}

File: 6th Semester/Programming Paradigms/NP2 Exercises - Part 2/ex24.py
# – Modifique o programa anterior de forma a ler um numero n. Imprima os n primeiros números primos.

def ehPrimo(num):
    i = 1
    first_do = True
    while (first_do or i <= (num / 2) + 1):
        first_do = False
        if ((i != 1 and num % i == 0) or num < 2 or (num != 2 and num % 2 == 0)):
            return False
        i += 2
    return True

num = int(input('Digite até quanto se quer conferir os números primos: '))
num = -num if num < 0 else num
for i in range(num):
    print(f'{i} ' if ehPrimo(i) else '', end='')
